src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -131,11 +154,29 @@ protected override void PostPersist(ApplicationEngine engine)
 
             // Distribute GAS for committee
 
-            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)ProtocolSettings.Default.CommitteeMembersCount);
+            int m = ProtocolSettings.Default.CommitteeMembersCount;
+            int n = ProtocolSettings.Default.ValidatorsCount;
+            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)m);
             var gasPerBlock = GetGasPerBlock(engine.Snapshot);
-            var pubkey = GetCommittee(engine.Snapshot)[index];
+            var committee = GetCommittee(engine.Snapshot);
+            var pubkey = committee[index];
             var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
             GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
+
+            // Record the cumulative reward of the voters of committee
+
+            var factor = index < n ? 2 : 1;
```

Comment:
The members returned by `GetCommittee()` are sorted by public key, not by votes. So you cannot simply check `index < n`. You must check whether it is a consensus node.

Reply:
Since we already have epoch, we don't need to sort committe by public key, just votes is enough.

Reply:
But `GetCommittee()` is used in other places.

Reply:
Only `GetCommitteeAddress` call it, we can sort it in `GetCommitteeAddress` method.

Reply:
I changed it to use `GetCommitteeFromCache()`.